public/app/features/explore/TraceView/components/TracePageHeader/SearchBar/NextPrevResult.tsx

```diff
@@ -0,0 +1,194 @@
+        grafana_version: config.buildInfo.version,
+        direction: 'next',
+      });
+
+      // new query || at end, go to start
+      if (
+        focusedSpanIndexForSearch === -1 ||
+        (spanFilterMatches && focusedSpanIndexForSearch === spanFilterMatches.size - 1)
+      ) {
+        setFocusedSpanIndexForSearch(0);
+        return;
+      }
+
+      // get next
+      setFocusedSpanIndexForSearch(focusedSpanIndexForSearch + 1);
+    }
+  };
+
+  const prevResult = (event: React.UIEvent, buttonEnabled: boolean) => {
+    event.preventDefault();
+    event.stopPropagation();
+
+    if (buttonEnabled) {
+      reportInteraction('grafana_traces_trace_view_find_next_prev_clicked', {
+        datasourceType: datasourceType,
+        grafana_version: config.buildInfo.version,
+        direction: 'prev',
+      });
+
+      // new query || at start, go to end
+      if (spanFilterMatches && (focusedSpanIndexForSearch === -1 || focusedSpanIndexForSearch === 0)) {
+        setFocusedSpanIndexForSearch(spanFilterMatches.size - 1);
+        return;
+      }
+
+      // get prev
+      setFocusedSpanIndexForSearch(focusedSpanIndexForSearch - 1);
+    }
+  };
+
+  const nextResultOnKeyDown = (event: React.KeyboardEvent, buttonEnabled: boolean) => {
+    if (event.key === 'Enter') {
+      nextResult(event, buttonEnabled);
+    }
+  };
+
+  const prevResultOnKeyDown = (event: React.KeyboardEvent, buttonEnabled: boolean) => {
+    if (event.key === 'Enter') {
+      prevResult(event, buttonEnabled);
+    }
+  };
+
+  const buttonEnabled = (spanFilterMatches && spanFilterMatches?.size > 0) ?? false;
+  const amountText = spanFilterMatches?.size === 1 ? 'match' : 'matches';
+  const matches =
+    spanFilterMatches?.size === 0 ? (
+      <>
+        <span>0 matches</span>
+        <Tooltip
+          content="There are 0 span matches for the filters selected. Please try removing some of the selected filters."
+          placement="left"
+        >
+          <span className={styles.matchesTooltip}>
+            <Icon name="info-circle" size="lg" />
+          </span>
+        </Tooltip>
+      </>
+    ) : focusedSpanIndexForSearch !== -1 ? (
+      `${focusedSpanIndexForSearch + 1}/${spanFilterMatches?.size} ${amountText}`
+    ) : (
+      `${spanFilterMatches?.size} ${amountText}`
+    );
+  const buttonClass = buttonEnabled ? styles.button : cx(styles.button, styles.buttonDisabled);
+
+  return (
+    <>
+      <span className={styles.matches}>{spanFilterMatches ? matches : `${totalSpans} spans`}</span>
+      <div className={buttonEnabled ? styles.buttons : cx(styles.buttons, styles.buttonsDisabled)}>
+        <div
+          aria-label="Prev result button"
+          className={buttonClass}
+          onClick={(event) => prevResult(event, buttonEnabled)}
+          onKeyDown={(event) => prevResultOnKeyDown(event, buttonEnabled)}
+          role="button"
+          tabIndex={buttonEnabled ? 0 : -1}
+        >
```

Comment:
Using `<div />` as this cannot be a nested button because it is to be used on the collapse bar header, which is already a button.